src/simulation/hdf_simulation_store.jl

```diff
@@ -699,16 +699,23 @@ function _read_length(::Type{OptimizerStats}, store::HdfSimulationStore)
     return HDF5.read(HDF5.attributes(dataset), "columns")
 end
 
-function _write_dataset!(dataset, array, row_range)
-    if ndims(array) == 2
-        dataset[:, 1, row_range] = array
-    elseif ndims(array) == 3
-        dataset[:, :, row_range] = array
-        #elseif ndims(array) == 4
-        #    dataset[:, :, :, row_range] = array
-    else
-        error("ndims not supported: $(ndims(array))")
-    end
+function _write_dataset!(dataset, array::Array{Float64, 2}, row_range::UnitRange{Int64})
+    dataset[:, 1, row_range] = array
+    @debug "wrote dataset" dataset row_range
+    return
+end
 
+function _write_dataset!(dataset, array::Array{Float64, 3}, row_range::UnitRange{Int64})
+    dataset[:, :, row_range] = array
     @debug "wrote dataset" dataset row_range
+    return
+end
+
+function _write_dataset!(
+    dataset,
+    ::Array{Float64, N},
+    row_range::UnitRange{Int64},
+) where {N}
+    error("ndims not supported: $N")
```

Comment:
ndims? Why not leave this unimplemented?

Reply:
I thought It would be better to Throw an error, I am fine removing it

Reply:
That is what I started doing. We had a team discussion and decided that the most "julian" thing to do is not implemented the fall-through method.

Reply:
ok, removing it